lib/rubocop/cop/rspec/prefer_before_over_setup.rb

```diff
@@ -0,0 +1,44 @@
+# frozen_string_literal: true
+
+module RuboCop
+  module Cop
+    module RSpec
+      # Checks that tests use rspec's `before` over rspec-rails' `setup` method.
+      #
+      # @example
+      #
+      #   # bad
+      #   setup do
+      #     allow(foo).to receive(:bar)
+      #   end
+      #
+      #   # good
+      #   before do
+      #     allow(foo).to receive(:bar)
+      #   end
+      #
+      class PreferBeforeOverSetup < Base
+        extend AutoCorrector
+
+        MSG = 'Use `before` instead of `setup`.'
+        SPACE = ' '
+
+        def_node_matcher :setup_call?, <<-PATTERN
+          (block
+            (send _ :setup)
+            (args) _)
+        PATTERN
+
+        def on_block(node)
+          return unless setup_call?(node)
+
+          add_offense(node) do |corrector|
+            setup_block_parts = node.source.split(/ /)
```

Comment:
I believe if you [captured](https://docs.rubocop.org/rubocop-ast/node_pattern.html#for-captures) the `:setup` with `$:setup`, you could just `corrector.replace(setup, 'before'`).

Reply:
@pirj For it to work you need to capture the whole node (`$:setup` captures `:send` symbol), anyway thanks for the tip!